File: WebDriverTests_Java/WebDriverTests/src/test/java/core/supportClass/BY.java
package core.supportClass;

import core.systemControls.HtmlControl;
import org.openqa.selenium.By;

public final class BY
{
    public static By Self(HtmlControl control)
    {
        return new BySelf(control);
    }


    public static By NthOfClass(String className, int index)
    {
        return new ByNthOfClass(className, index);
    }

    public static By ByNthOfBy(By byName, int index)
    {
        return new ByNthOfBy(byName, index);
    }


    public static By ByDataUnique(String byName)
    {
        return new ByDataUnique(byName);
    }
}

File: WebDriverTests_Csharp/Autotest.Core/Autotests.Utilities/WebTestCore/TestSystem/BY.cs
using System;
using Autotests.Utilities.WebTestCore.SystemControls;
using Autotests.Utilities.WebTestCore.TestSystem.ByExtensions;
using Autotests.Utilities.WebTestCore.TestSystem.ByExtensions.ChangeId;
using OpenQA.Selenium;

namespace Autotests.Utilities.WebTestCore.TestSystem
{
    public static class BY
    {
        public static By Self(HtmlControl control)
        {
            return new BySelf(control);
        }

        public static By ByAndManualCondition(By by, Func<IWebElement, bool> condition)
        {
            return new ByByAndManualCondition(by, condition);
        }

        public static By NthOfBy(By by, int index)
        {
            return new ByNthOfBy(by, index);
        }

        public static By NthOfClass(string className, int index)
        {
            return new ByNthOfClass(className, index);
        }

        public static By ModifiedIdOfHtmlContol(HtmlControl control, ChangeIdStrategy changeIdStrategy)
        {
            return new ByModifiedIdOfHtmlContol(control, changeIdStrategy);
        }
    }
}